drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/create_referral_job.rb

```diff
@@ -94,9 +99,65 @@ def create_client(attrs)
       client.veteran_status = 99
       client.ethnicity = 99
       client.save!
+
+      # additional attributes set if this client is the hoh
+      setup_hoh(client) if attrs[:relationship_to_hoh] == 1
+
       client
     end
 
+    def setup_hoh(client)
+      common_client_attrs = {
+        PersonalID: client.PersonalID,
+        UserID: client.UserID,
+        data_source_id: client.data_source_id,
+      }
+      client_address_attrs = params[:addresses].to_a.map do |values|
+        {
+          postal_code: values[:zip],
+          **values.slice(
+            :line1,
+            :line2,
+            :city,
+            :state,
+            :use,
+            # :county,  FIXME - spec doc has a "county" field; We don't have that but we do have "country" - maybe there's a typo somewhere?
+          ),
+          # FIXME: unsure what to do with this, use uuid for now
+          AddressID: bogus_id,
```

Comment:
Thats basicaly right, generate a uuid. That's what we do for all the HUD "ID" fields. But you can use this func which strips out dashes:
```suggestion
          AddressID: Hmis::Hud::Base.generate_uuid,
```